### PR TITLE
Poll complete status

### PR DESCRIPTION
**Pull request by filg68**
Makes the following changes:

-  Adds complete fields to Poll Model
-  Adds algorithm to voteModelUpdate to calculate completeness status
-  Updates addPoll module to add complete field on poll create
-  Updates seedDb to include complete poll when creating polls for seed data
-  Polls are now sorted by ( -complete,  -createdAt ) - oldest completed polls show up first, then oldest uncompleted

### CODE REVIEW

File: client/src/components/AddPollDialog.js [hunk 1]
```diff
@@ -90,6 +90,7 @@ class AddPollDialog extends Component {
             image2: "",
             expiresOn: "",
             sendToList: "",
+            selectedList: "",
```
Comment (IoPhoenix) [inline client/src/components/AddPollDialog.js]: this key is declared but not used anywhere

File: client/src/components/AddPollDialog.js [hunk 2]
```diff
@@ -121,6 +123,12 @@ class AddPollDialog extends Component {
 
     handleSelectChange = e => {
         this.setState({ sendToList: e.target.value });
```
Comment (IoPhoenix) [inline client/src/components/AddPollDialog.js]: As an alternative, you can use custom attribute on MenuItem: `<MenuItem data-title={list.title}>` and then access it as `e.currentTarget.dataset.title`

File: src/routes/api/users.js [hunk 3]
```diff
@@ -216,9 +216,14 @@ router.get("/user/:id", (req, res) => {
...
                 sort: {
+                    complete: "descending",
                     createdAt: "descending"
                 }
-            }
+            },
+            populate: {
+                path: "sendToList",
+                select: "title"
+            } // select only name and avatar for each friend
```
Comment (IoPhoenix) [inline src/routes/api/users.js]: the comment is irrelevant

File: client/src/components/AddPollDialog.js [hunk 4]
```diff
@@ -162,7 +170,13 @@ class AddPollDialog extends Component {
                         return;
                     }
 
-                    // add new poll to Profile and close dialog:
+                    // add the list title and voter counts to the returned data
+                    response.data.data.sendToList = {
```
Comment (IoPhoenix) [inline client/src/components/AddPollDialog.js]: Can we do something to simplify `response.data.data`? Looks weird...

File: client/src/components/AddPollDialog.js (hunk 1 above)
Comment (filg68) [reply client/src/components/AddPollDialog.js]: Removed it.

File: src/routes/api/users.js (hunk 3 above)
Comment (filg68) [reply src/routes/api/users.js]: Cut and paste error - removed it

File: client/src/components/AddPollDialog.js (hunk 4 above)
Comment (filg68) [reply client/src/components/AddPollDialog.js]: Changed key of returned data to newPoll ---> response.data.newPoll

File: client/src/components/AddPollDialog.js (hunk 2 above)
Comment (filg68) [reply client/src/components/AddPollDialog.js]: Noted.

**Pull request merged**